src/catalog/catalog_entry/table_catalog_entry.cpp

```diff
@@ -377,6 +585,73 @@ unique_ptr<CatalogEntry> TableCatalogEntry::SetDefault(ClientContext &context, S
 	return make_unique<TableCatalogEntry>(catalog, schema, (BoundCreateTableInfo *)bound_create_info.get(), storage);
 }
 
+unique_ptr<CatalogEntry> TableCatalogEntry::ChangeGeneratedColumnType(ClientContext &context,
+                                                                      ChangeColumnTypeInfo &info) {
+	auto change_info = GetColumnInfo(info.column_name);
```

Comment:
Perhaps changing a generated column type should not be supported and just throw an error, or it could be rewritten into a drop + create of a new expression with a cast pushed on top.

The reason changing a regular column is interesting/more difficult is because we want to change the type while preserving the data. For a generated column no data is stored, so changing the type is equivalent to dropping and re-creating the column with a cast added.

Reply:
Postgres allows the change, but the resulting behavior is kind of broken

```sql
CREATE TABLE orders_items (
	product_id INTEGER NOT NULL,
	quantity INTEGER,
	total_item_price INTEGER GENERATED ALWAYS AS (quantity * product_id) STORED
);

ALTER TABLE orders_items ALTER COLUMN total_item_price TYPE VARCHAR;

INSERT INTO orders_items (product_id) VALUES (5);

SELECT * FROM orders_items;
```
```
 product_id | quantity | total_item_price 
------------+----------+------------------
          5 |          | 
```

Reply:
Casts are only performed when there is data present, so detecting the error on insertion makes sense for the STORED kind.
For the VIRTUAL kind the expression is only materialized when a SELECT statement is performed, so that's the first time the conversion error gets detected